### PR TITLE
Support fork for any network

### PR DESCRIPTION
**Pull request by agAgent0xF8**

### CODE REVIEW

File: helpers/misc-utils.ts [hunk 1]
```diff
@@ -19,6 +19,28 @@ export const setDRE = (_DRE: HardhatRuntimeEnvironment) => {
   DRE = _DRE;
 };
 
+export const isForkNetwork = (): boolean => {
+  return process.env.FORK ? true : false;
```
Comment (agAgent013) [inline helpers/misc-utils.ts:23]: improvement: it's possible to use `!!` or `Boolean` to convert string value to boolean.

File: tasks/helpers/bsc-fork.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { eOtherNetwork } from '../../helpers/types';
+
+export const workaroundForBSCFork = async () => {
+  if (!isForkNetwork() || getNetworkName() != eOtherNetwork.bsc) {
+    return;
+  }
+
+  const url = NETWORKS_RPC_URL[eOtherNetwork.bsc];
+  const bscProvider = new ethers.providers.JsonRpcProvider(url);
+  let latestBlock = (await bscProvider.getBlockNumber()) + 5;
```
Comment (agAgent013) [inline tasks/helpers/bsc-fork.ts]: I'd use const because latestBlock isn't changed here.

File: tasks/helpers/bsc-fork.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      await DRE.network.provider.send('hardhat_reset', [
+        {
+          forking: {
+            jsonRpcUrl: url,
+            blockNumber: latestBlock,
+          },
+        },
+      ]);
+      return;
+    } catch (err: any) {
```
Comment (agAgent013) [inline tasks/helpers/bsc-fork.ts:29]: JFYI: it's possible to use global type Error.

File: helpers/misc-utils.ts (hunk 1 above)
Comment (agAgent0xF8) [reply helpers/misc-utils.ts:23]: ok, but I just find it more readable this way

File: tasks/helpers/bsc-fork.ts (hunk 3 above)
Comment (agAgent0xF8) [reply tasks/helpers/bsc-fork.ts:29]: sometimes there isn't Error at all

File: tasks/helpers/bsc-fork.ts (hunk 2 above)
Comment (agAgent0xF8) [reply tasks/helpers/bsc-fork.ts]: it was changing ;) fixed

**Review by agAgent013 — APPROVED** [review]

**Pull request merged**